File: src/main/kotlin/dev/jefy/connectpro/marketplace/applicaion/query/MarketPlaceListingQuery.java
package dev.jefy.connectpro.marketplace.applicaion.query;

import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.marketplace.applicaion.dtos.JobPostListingResponse;
import dev.jefy.connectpro.marketplace.applicaion.dtos.SearchRequest;
import dev.jefy.connectpro.marketplace.applicaion.dtos.ServiceListingResponse;
import dev.jefy.connectpro.shared.application.dtos.PageResponse;

/**
 * @author Jôph E.F. Yamba
 * @date 08/04/2026
 * @time 11:48:26
 */
/**
 * @author Jôph Yamba
 */
@NullMarked
public interface MarketPlaceListingQuery {
    
    PageResponse<ServiceListingResponse> filterService(SearchRequest request);
    PageResponse<JobPostListingResponse> filterJobPost(SearchRequest request);

    PageResponse<ServiceListingResponse> getRecommendedServices(int page, int size);
    PageResponse<JobPostListingResponse> getRecommendedJobPosts(int page, int size);
}

File: src/main/kotlin/dev/jefy/connectpro/marketplace/application/query/MarketPlaceListingQuery.kt
package dev.jefy.connectpro.marketplace.application.query

import dev.jefy.connectpro.marketplace.application.dtos.JobPostListingResponse
import dev.jefy.connectpro.marketplace.application.dtos.SearchRequest
import dev.jefy.connectpro.marketplace.application.dtos.ServiceListingResponse
import dev.jefy.connectpro.shared.application.dtos.PageResponse
import org.jspecify.annotations.NullMarked

/**
 * @author Jôph Yamba
 */
@NullMarked
interface MarketPlaceListingQuery {
    fun filterService(request: SearchRequest): PageResponse<ServiceListingResponse>
    fun filterJobPost(request: SearchRequest): PageResponse<JobPostListingResponse>

    fun getRecommendedServices(page: Int, size: Int): PageResponse<ServiceListingResponse>
    fun getRecommendedJobPosts(page: Int, size: Int): PageResponse<JobPostListingResponse>
}
